### ISSUE TITLE
Expose captured groups for regex match

### ISSUE THREAD

**Issue by Luro02** [maciejhirsz/logos#143]

**Comment by maciejhirsz** [comment]
Logos isn't using the `regex` crate, and it currently has no notion of captured groups in the state machine.

This would be possible to do as an opt-in by passing captures into the callback as an optional parameter with an extra flag in the `#[regex]` attribute:

```rust
#[derive(Logos)]
enum Token {
    #[regex("foo(bar)". foobar, captures = true)]
    FooBar,

    // ...
}

fn foobar(lex: &mut Lexer<Token>, captures: &logos::Captures) {
    assert_eq!(captures.get(0), Some("bar"));
}
```

I have some idea how this could be implemented.

**Comment by jeertmans** [comment]
Hello @maciejhirsz , do you plan to release such a feature in the near future ? Thanks :)

**Comment by CraftSpider** [comment]
I could use a feature like this to simplify my handling of things like prefixed integer tokens

**Comment by ModProg** [comment]
This would also interresting for e.g. a doc comment that does not want to match the comment symbol it self:
```rs
    #[regex(r"///([^/\n][^\n]*)?\n")]
    DocComment(&'source str),
```
I only want the body neither `///` nor `\n`

**Comment by ModProg** [comment]
> @ModProg you can do it trivially by slicing the match in a callback

Yes, that's how I did it, I only thought about having a way to say store this match in the Variant. (with named capture groups you could even do named fields)
